ebi_archive/lib/downloader.rs

```diff
@@ -0,0 +1,72 @@
+use crate::errors::download::DownloadError;
+use crate::Result;
+use ebi_sources::source::Source;
+use futures::future::join_all;
+use reqwest;
+use std::fs;
+use std::fs::File;
+use std::io;
+use std::io::Cursor;
+use tokio::task::JoinHandle;
+
+pub async fn download_chapter(
+    source: Box<dyn Source>,
+    manga_identifier: &str,
+    chapter_number: usize,
+) -> Result<()> {
+    let created_directory_result = fs::create_dir_all(format!(
+        "{source_name}/{identifier}/{chapter}",
+        source_name = "opex",
+        identifier = manga_identifier,
+        chapter = chapter_number
+    ));
+
+    let _created_directory = match created_directory_result {
+        Ok(directory) => directory,
+        Err(error) => panic!("Problem creating chapter directory: {}", error),
+    };
```

Comment:
I feel like create_directory could be a separated function

Reply:
Either way, avoid panic: error propagation is your friend.
Everything thing could be simplified to:

```suggestion
    let _= fs::create_dir_all(format!(
        "{source_name}/{identifier}/{chapter}",
        source_name = "opex",
        identifier = manga_identifier,
        chapter = chapter_number
    ))?;
```